crates/amaru-kernel/src/serde_utils.rs

```diff
@@ -65,7 +71,39 @@ impl From<TransactionOutputProxy> for TransactionOutput {
             address: proxy.address,
             value: Value::Coin(proxy.value.unwrap_or_default()),
             datum_option: None,
-            script_ref: None,
+            script_ref: proxy
+                .script_ref
+                .map(|proxy| CborWrap(ScriptRef::from(proxy))),
         })
     }
 }
+
+// ------------------------------------------------------------------------------- TransactionOutput
+
+#[derive(Debug, serde::Deserialize)]
+pub enum ScriptRefProxy {
+    NativeScript(Bytes),
+    PlutusV1(Bytes),
+    PlutusV2(Bytes),
+    PlutusV3(Bytes),
+}
+
+impl HasProxy for ScriptRef {
+    type Proxy = ScriptRefProxy;
+}
+
+impl From<ScriptRefProxy> for ScriptRef {
+    // TODO: Is there a better way to do this? Just brute force to get it working for now...
+    #[allow(clippy::unwrap_used)]
```

Comment:
If it can fail, would be better to return a `Result` and handle it down the stream.

Reply:
It's only something that ever runs in tests and it *should* as it's decoding test data which we've set up manually.